src/server/game/Autobroadcast/AutobroadcastMgr.cpp

```diff
@@ -101,42 +133,105 @@ void AutobroadcastMgr::SendAutobroadcasts()
+        SendWorldAnnouncement(textId);
+        SendNotificationAnnouncement(textId);
     default:
         break;
     }
 
-    LOG_DEBUG("autobroadcast", "AutobroadcastMgr::SendAutobroadcasts: '{}'", msg);
+    LOG_DEBUG("autobroadcast", "AutobroadcastMgr::SendAutobroadcasts: '{}'", textId);
 }
 
-void AutobroadcastMgr::SendWorldAnnouncement(std::string msg)
+void AutobroadcastMgr::SendWorldAnnouncement(uint8 textId)
 {
-    ChatHandler(nullptr).SendWorldTextOptional(LANG_AUTO_BROADCAST, ANNOUNCER_FLAG_DISABLE_AUTOBROADCAST, msg.data());
+    // Send localized messages to all sessions
+    ChatHandler(nullptr).DoForAllValidSessions([&](Player* player)
+        {
+            if (!player || !player->GetSession())
+            return;
+
+    // Get player's locale
+    LocaleConstant locale = player->GetSession()->GetSessionDbLocaleIndex();
+
+    std::string localizedMessage;
+
+    if (_localizedAutobroadcasts.count(textId) && _localizedAutobroadcasts[textId].count(locale))
+    {
+        // Use the localized message if available
+        localizedMessage = _localizedAutobroadcasts[textId][locale];
+    }
+    else if (_autobroadcasts.count(textId))
+    {
+        // Fallback to the default message if localization is unavailable
+        localizedMessage = _autobroadcasts[textId];
+    }
+    else
+    {
+        // Fallback message if textId doesn't exist
+        localizedMessage = "Broadcast Message Not Found";
+    }
+
+    // Send the localized or fallback message
+    ChatHandler(player->GetSession()).SendWorldTextOptional(localizedMessage, ANNOUNCER_FLAG_DISABLE_AUTOBROADCAST);
+        });
 }
 
-void AutobroadcastMgr::SendNotificationAnnouncement(std::string msg)
+void AutobroadcastMgr::SendNotificationAnnouncement(uint8 textId)
 {
-    WorldPacket data(SMSG_NOTIFICATION, (msg.size() + 1));
-    data << msg.data();
-    sWorld->SendGlobalMessage(&data);
+    for (auto const& sessionPair : sWorld->GetAllSessions())
+    {
+        WorldSession* session = sessionPair.second;
+
+        if (session &&
+            session->GetPlayer() &&
+            session->GetPlayer()->IsInWorld())
+        {
+            // Retrieve player's locale
+            LocaleConstant locale = session->GetSessionDbLocaleIndex();
+
+            // Lookup the localized message based on textId and locale
+            std::string localizedMessage;
+
+            if (_localizedAutobroadcasts.count(textId) &&
+                _localizedAutobroadcasts[textId].count(locale))
+            {
+                // Use localized message
+                localizedMessage = _localizedAutobroadcasts[textId][locale];
+            }
+            else
+            {
+                // Fallback to default locale or placeholder text
+                if (_autobroadcasts.count(textId))
+                    localizedMessage = _autobroadcasts[textId];
+                else
+                    localizedMessage = "Broadcast Message Not Found";
```

Comment:
it should probably fall back to the og text.

Reply:
There might be a fallback for the og message, if it doesnt exist. Technically this should never happen as the ids are selected from the ids derived from the base auto broadcast table.

Reply:
If there is no OG message we should error at startup imo